### ISSUE TITLE
Improvement to handle irregular strings as keyname.

### ISSUE THREAD

**Issue by ksauzz** [basho/riak_cs#910]
This is API compatibility issue. If one uses `test%%00` for keyname:
- S3 handles this as `test%%00`
- CS handles this as `test%0x00`

Then, `s3cmd ls s3://bucketname` fails due to XML response including `<Key>hoge%\x00</Key>`.

**Comment by shino** [comment]
It's useful (for searchability) to copy the stack on this error from #974.

```
01:01:04.857 [error] Lager event handler error_logger_lager_h exited with reason
 {'EXIT',{{badmatch,["/buckets/test/objects",{error,{error,function_clause,
   [{xmerl_lib,expand_attributes,["abc",1,[{error,1},{'Key',1},{'Contents',16},{'ListBucketResult',1}]],
    [{file,"src/xmerl_lib.erl"},{line,234}]},{xmerl_lib,expand_element,4,[{file,"src/xmerl_lib.erl"},{line,179}]},{xmerl_lib,expand_content,4,[{file,"src/xmerl_lib.erl"},{line,224}]},{xmerl_lib,expand_element,4,[{file,"src/xmerl_lib.erl"},{line,187}]},{xmerl_lib,expand_content,4,[{file,"src/xmerl_lib.erl"},{line,224}]},{xmerl_lib,...},...]}},...]},...}}
```

And another possibility, `xmerl_lib:expand_element` on top of stack.

```
10:00:44.885 [error] Lager event handler error_logger_lager_h exited with reason
 {'EXIT',{{badmatch,["/buckets/test2/objects",{error,{error,function_clause,
   [{xmerl_lib,expand_content,[<<"ÿ">>,1,[{error,1},{'Key',1},{'Contents',7},{'ListBucketResult',1}],false],
    [{file,"src/xmerl_lib.erl"},{line,216}]},{xmerl_lib,expand_element,4,[{file,"src/xmerl_lib.erl"},{line,180}]},{xmerl_lib,expand_content,4,[{file,"src/xmerl_lib.erl"},{line,224}]},{xmerl_lib,expand_element,4,[{file,"src/xmerl_lib.erl"},{line,187}]},{xmerl_lib,expand_content,4,[{file,"src/xmerl_lib.erl"},{line,224}]},...]}},...]},...}}
```

**Comment by shino** [comment]
Comment about #977.

This error typically happen when key contains non-UTF-8 byte sequence, but it is possible for ASCII-only keys are affected by the bug.

**Comment by ksauzz** [comment]
Fixed by #1040
